uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/leave-request.controller.js

```diff
@@ -5,23 +5,48 @@ define([
 ], function (controllers) {
   controllers.controller('LeaveRequestCtrl', LeaveRequestCtrl);
 
-  LeaveRequestCtrl.$inject = ['$log', '$q', 'parentCtrl'];
+  LeaveRequestCtrl.$inject = ['$controller', '$log', '$q', 'parentCtrl'];
 
-  function LeaveRequestCtrl ($log, $q, parentCtrl) {
+  function LeaveRequestCtrl ($controller, $log, $q, parentCtrl) {
     $log.debug('LeaveRequestCtrl');
 
-    var vm = parentCtrl;
+    parentCtrl.calculateBalanceChange = calculateBalanceChange;
+    parentCtrl.canCalculateChange = canCalculateChange;
+    parentCtrl.checkSubmitConditions = checkSubmitConditions;
+    parentCtrl.initChildController = initChildController;
 
-    vm.checkSubmitConditions = checkSubmitConditions;
-    vm.initChildController = initChildController;
+    /**
+     * Calculates balance change
+     *
+     * @return {Promise}
+     */
+    function calculateBalanceChange () {
+      return parentCtrl.request.calculateBalanceChange(parentCtrl.selectedAbsenceType.calculation_unit_name);
+    }
+
+    /**
+     * Checks if change can be calculated
```

Comment:
describes the conditions

Reply:
@AkA84 sure. I also augmented the function. I also included the check for hours type (requests in hours require to have deductions). It worked earlier because deductions are set on *watching* dates, but it would be safer to have that extra condition so it doesn't depend on the implementation of the flow.